marklogic-data-hub/src/test/java/com/marklogic/hub/security/DataHubSecurityAdminTest.java

```diff
@@ -271,4 +272,28 @@ void task5CreateRoleWithCustomPrivilegeInDisallowedNamespace() {
             new Role(adminUserApi, roleName).delete();
         }
     }
+
+    @Test
+    void CreateCustomRoleWithXdmpShutdown() {
+        MarkLogicVersion version = new MarkLogicVersion(getHubClient().getManageClient());
+        Assumptions.assumeTrue(version.getMajor() >= 10 &&
+            (version.isNightly() || version.getMinor() >= 601), "Bug<55765> is fixed in 10.0-6.1");
+
+        final String roleName = "custom-role-with-xdmp-shutdown-privilege";
+        Role customRole = new Role(userWithRoleBeingTestedApi, roleName);
+        customRole.addPrivilege(new RolePrivilege("xdmp:shutdown", "http://marklogic.com/xdmp/privileges/xdmp-shutdown", "execute"));
+
+        try {
+            customRole.save();
+            fail("The role creation should have failed because a DHSA user shouldn’t be able to create a custom role with xdmp:shutdown privilege");
+        }
+        catch (Exception ex) {
+            logger.info("Caught expected exception: " + ex.getMessage());
+
+            // Just in case the role was created (role was being created even after exception thrown -> bug: 55765).
+            assertNull(customRole.getRole());
```

Comment:
This is a good check to make, but what this is currently doing is verifying that the customRole doesn't inherit any roles. What we want to do instead is verify that the role was not created:

```
assertFalse(customRole.exists(), "The role should not have been created");
```

The exists() call will under the hood make a call to the Manage API to see if the resource exists or not.

Reply:
Thanks Rob! This is what I was intending to do.